test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -1746,6 +2151,208 @@ public void UpToDate_InvalidOptions()
             .And.HaveStdErrContaining(string.Format(CliCommandStrings.InvalidOptionCombination, RunCommandParser.NoCacheOption.Name, RunCommandParser.NoBuildOption.Name));
     }
 
+    [Fact]
+    public void CscOnly()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory(baseDirectory: OutOfTreeBaseDirectory);
+
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.cs"), """
+            Console.WriteLine("v1");
+            """);
+
+        // Remove artifacts from possible previous runs of this test.
```

Comment:
Roslyn deals with this kind of thing by generating a unique temp subdirectory, doing the I/O in there, then deleting it at the end of the test. It might be reasonable to also do that here. However, I don't have any strong objection to this way, especially because I do not think we plan on parallelizing these tests.

Reply:
The tests don't control where the artifacts are placed though (they could but then they wouldn't test the real product as much). The artifacts path comes from our algorithm to hash the full source path (which is in a temp dir unique to the test) and put it under a temp subdirectory. Which also means the artifacts should be placed at unique location per each test, so parallelization shouldn't be a problem.

Reply:
> which is in a temp dir unique to the test

Ah, I must have missed the part of the code which is doing this part. Thanks.